tests/integration/high_availability/test_replication.py

```diff
@@ -85,131 +54,82 @@ async def test_consistent_data_replication_across_cluster(
     ops_test: OpsTest,
 ) -> None:
     """Confirm that data is replicated from the primary node to all the replicas."""
-    # Insert values into a table on the primary unit
-    app = await app_name(ops_test)
-    random_unit = ops_test.model.applications[app].units[0]
-    cluster = cluster_name(random_unit, ops_test.model.info.name)
-    server_config_credentials = await get_server_config_credentials(random_unit)
+    mysql_application_name, _ = await high_availability_test_setup(ops_test)
 
-    primary_unit = await get_primary_unit(
-        ops_test,
-        random_unit,
-        app,
-        cluster,
-        server_config_credentials["username"],
-        server_config_credentials["password"],
-    )
-    primary_unit_address = await primary_unit.get_public_address()
+    # assert that there are 3 units in the mysql cluster
+    assert len(ops_test.model.applications[mysql_application_name].units) == 3
 
-    random_chars = generate_random_string(40)
-    create_records_sql = [
-        "CREATE DATABASE IF NOT EXISTS test",
-        "CREATE TABLE IF NOT EXISTS test.data_replication_table (id varchar(40), primary key(id))",
-        f"INSERT INTO test.data_replication_table VALUES ('{random_chars}')",
-    ]
+    database_name, table_name = "test-check-consistency", "data"
+    await insert_data_into_mysql_and_validate_replication(ops_test, database_name, table_name)
+    await clean_up_database_and_table(ops_test, database_name, table_name)
```

Comment:
Would it be a good idea to move this to a fixture to cleanup the database and table if the `insert_data_into_mysql_and_validate_replication ` function fails?